tests/SAML2/XML/alg/DigestMethodTest.php

```diff
@@ -15,25 +15,30 @@
  * @author Jaime Pérez Crespo, UNINETT AS <jaime.perez@uninett.no>
  * @package simplesamlphp/saml2
  */
-class DigestMethodTest extends \PHPUnit\Framework\TestCase
+final class DigestMethodTest extends \PHPUnit\Framework\TestCase
 {
+    /** @var \DOMDocument */
+    private $document;
+
+
     /**
      * @return void
      */
-    public function testMarshalling(): void
+    protected function setUp(): void
     {
-        $digestMethod = new DigestMethod('http://exampleAlgorithm');
+        $this->document = DOMDocumentFactory::fromString(
+            '<alg:DigestMethod xmlns:alg="' . DigestMethod::NS . '" Algorithm="http://exampleAlgorithm" />'
+        );
+    }
 
-        $document = DOMDocumentFactory::fromString('<root />');
-        $xml = $digestMethod->toXML($document->firstChild);
 
-        $digestMethodElements = Utils::xpQuery(
-            $xml,
-            '/root/*[local-name()=\'DigestMethod\' and namespace-uri()=\'' . DigestMethod::NS . '\']'
-        );
-        $this->assertCount(1, $digestMethodElements);
-        $digestMethodElement = $digestMethodElements[0];
-        $this->assertEquals('http://exampleAlgorithm', $digestMethodElement->getAttribute('Algorithm'));
+    /**
+     * @return void
+     */
+    public function testMarshalling(): void
+    {
+        $digestMethod = new DigestMethod('http://exampleAlgorithm');
+        $this->assertEquals($this->document->saveXML($this->document->documentElement), strval($digestMethod));
```

Comment:
I typically add tests for all the getters in the object, so that we not only test that `toXML()` works fine, but also that we can extract the information properly from the object.